### DISCUSSION TITLE
Configuration input.path remote files?

### DISCUSSION THREAD

**Discussion by wouterkroes** [orval-labs/orval#108] [Ideas]
Let me start with: thanks for this great library!

If I am not mistaken the configuration `input.path` (https://orval.dev/reference/configuration#target) only supports local files. Would it be possible to support an URL to an external Open API definition as `input.path`?

**Comment by anymaniax** [comment]
Hello,

Thanks for your interest! And don't hesitate if you have new idea.

Normally, it's already the case. And it's also work with a github link too a private repository.

**Comment by wouterkroes** [comment]
Thanks for the quick reply!

Hmm, I was under the assumption that this was not supported because I tried it locally.

I am probably doing something wrong because:

**Situation 1**

Local URL without `www`

```
orval.config.js

module.exports = {
  'petstore-file': {
    input: 'http://localhost:3001/users',
    output: './src/petstore.ts',
  },
};
```

I get the following error:

```
Error: ENOENT: no such file or directory, open '/Users/.../http:/localhost:3001/users'
```

**Situation 2**

Local URL with `www`

```
orval.config.js

module.exports = {
  'petstore-file': {
    input: 'http://www.localhost:3001/users',
    output: './src/petstore.ts',
  },
};
```

I get the following error:

```
> orval

🍻 Start orval v4.1.0 - A swagger client generator for typescript
Oups... 🍻
```

**Situation 3**

Github link too a public repository

```
orval.config.js

module.exports = {
  'petstore-file': {
    input:
      'https://raw.githubusercontent.com/anymaniax/orval/master/samples/basic/petstore.yaml',
    output: './src/petstore.ts',
  },
};
```

I get the following error:
```
> orval

🍻 Start orval v4.1.0 - A swagger client generator for typescript
Oups... 🍻
```

If it makes more sense to move this chat to the "issues" section, let me know

**Comment by anymaniax** [comment]
Hey,

I published a new version which supports the content type `text/plain`. You just need to install orval@next. Now, the situation 3 should work. For the situation 1 and 2, I need more information which content type do you return? Application/json or text/plain?

**Comment by wouterkroes** [comment]
Hey,

Adding `Content-Type`:`text/plain` fixes situation 3 (github link) 👍 

But sadly it does not fixes situation 1 + 2 (local files)
My local server is [Serve](https://www.npmjs.com/package/serve).
Serve returns the `Content-Type`: `text/yaml` (URL http://xxx.xxx.x.xx:xxxx/petstore.yml).

So I guess the fix is adding `text/yaml` to `lib/core/importers/specs.js` -> https://github.com/anymaniax/orval/pull/109?

**Reply by anymaniax** [reply]
Yes and also set correctly the format const in the same file because normally I get the file format from the input path. For application/json and text/yaml, it's easy we can set directly the format. But when we receive a text/plain, we need to check the content of the payload. I would just check if the string `.startsWith('{')` to know if it's a JSON specification.

**Reply by wouterkroes** [reply]
I have updated the PR based on your reply. Hopefully, this is what you had in mind? If not, please respond in the PR 👍
